h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -4048,7 +4066,12 @@ public BetaConstraint(Frame beta_constraints) {
           }
         }
       }
-      if (_parms._non_negative) setNonNegative();
+      if (_parms._non_negative) {
```

Comment:
If I understand the changes to the code correctly, are we setting the non_negative parameter to None if splines_non_negative is set by user for monotone splines?

Reply:
@narasimhard:

GLM has been used to build other models and one of them is GAM.  So, if the user sets non_negative parameter, the actions we take will depend on which glmType we are encountering.  If the glmType is GAM, we will let the beta constraints that are already specified in the beta_constraint frame take precedence.  That is why the method setNonNegative is called with the frame as the method input.  If you look at setNonNegative(beta_constraints), it will not set the non-negative for any predictor that is already specified in the beta_constraints frame.  It will only set non-negative to predictors that are not specified in the beta_constraints frame.  If the glmType is not GAM, then, it will be business as usual as before.

Reply:
Sounds good, thanks! If the glmType is GAM, then non-negative is set for predictor variables rather than GAM columns(assuming beta_constraints as splines_non_negative for GAM cols).